src/App.tsx

```diff
@@ -4,25 +4,25 @@ import { useEffect, useState } from "react";
 import Confetti from "react-confetti";
 import { PizzaInterface } from "./components/pizza";
 import Wheel from "./components/wheel";
+import { useQuery } from "react-query";
+
+type PizzaCategory = "dessert" | "original" | "tynn" | "vegansk";
+const pizzaCategoryNames: PizzaCategory[] = ["dessert", "original", "tynn", "vegansk"];
 
 function App() {
   const apiURL = "https://pizzapicker-api.vercel.app/";
 
-  const [pizzas, setPizzas] = useState<{
-    dessert: PizzaInterface[];
-    original: PizzaInterface[];
-    tynn: PizzaInterface[];
-    vegansk: PizzaInterface[];
-  }>({ dessert: [], original: [], tynn: [], vegansk: [] });
+  const { data: pizzaCategories, isLoading, error } = useQuery("pizzas", async () =>
+    (await axios.get<Record<PizzaCategory, PizzaInterface[]>>(apiURL)).data
+  );
+
+  const [activeCategories, setActiveCategories] = useState<PizzaCategory[]>(pizzaCategoryNames);
 
-  const [activeCatergories, setActiveCatergories] = useState([
-    "original",
-    "tynn",
-    "vegansk",
-    "dessert",
-  ]);
+  const pizzaOptions = pizzaCategoryNames
+    .filter(category => activeCategories.includes(category)) 
+    .map(category => pizzaCategories?.[category])
+    .flatMap(pizzas => pizzas ?? []);
```

Comment:
pizza.type should be set to category . otherwise the custom pizza images wont work (all pizzas are currently set to default)

`pizza.tsx Line:15 to Line:23`:
```
const Pizza: React.FC<Props> = ({ pizza }) => {
  const pizzaType =
    pizza.type === "dessert"
      ? style.candy
      : pizza.type === "tynn"
      ? style.thin
      : pizza.type === "vegansk"
      ? style.vegan
      : style.regular;

```
this code can probably also be improved lol